src/core/utils/workspace.test.ts

```diff
@@ -8,11 +8,11 @@ import { httpServiceMock } from '../public/mocks';
 
 describe('#getWorkspaceIdFromUrl', () => {
   it('return workspace when there is a match', () => {
-    expect(getWorkspaceIdFromUrl('http://localhost/w/foo')).toEqual('foo');
+    expect(getWorkspaceIdFromUrl('http://localhost/w/foo', '')).toEqual('foo');
```

Comment:
Q: why it need to specify `''` as the second parameter explicitly?

Reply:
Maybe my question is more about why we need to change the `getWorkspaceIdFromUrl` function signature?

Reply:
1. From the implementation perspective, the `basePath` parameter is required when constructing the regular expression to parse the workspace id from url.
2. The bug comes from the loose function signature, I change it to a more strict signature so that others won't step into similar bug.